pilot/pkg/model/virtualservice.go

```diff
@@ -429,9 +429,10 @@ func stringMatchConflict(root, leaf *networking.StringMatch) bool {
 	if root == nil || leaf == nil {
 		return false
 	}
-	// regex match is not allowed
+	// TODO: check regex conflict
 	if root.GetRegex() != "" || leaf.GetRegex() != "" {
-		return true
+		log.Warnf("It is a risk of conflict that either root(%s) or delegate(%s) have regex set, "+
```

Comment:
This is not an appropriate place to log, it is not visible to user applying the virtual service and will be very spammy since this is computed many many times

Reply:
> This is not an appropriate place to log, it is not visible to user applying the virtual service and will be very spammy since this is computed many many times

i will change the log level to debug and warn the user when the mergeHTTPRoutes function  executes